### PR TITLE
Rework kubeconfig path generation for cli

### PR DESCRIPTION
**Pull request by chrisdoherty4**
Rework kubeconfig file path creation to decouple it and make error messages consistent.

The kubeconfig path was constructed in several places within the code. This meant changing the structure would result in multiple changes. Additionally, there was struct contamination likely stemming from copy-paste that resulted in structs reaching into other structs to call their unexported methods for overriding kubeconfig's with a user defined kubeconfig. While all the cmd structs are in 1 package, this is still bad practice so it has been replaced with a piece reusable logic and the cmds completely separated. 

Interestingly, some of the diagnostic related cmds had single character differences making it extremely hard to tell they were calling each other.

### CODE REVIEW

**Review by TerryHowe — APPROVED** [review]

File: cmd/eksctl-anywhere/cmd/createcluster.go [hunk 1]
```diff
@@ -74,15 +76,20 @@ func (cc *createClusterOptions) validate(ctx context.Context) error {
...
+		return kubeconfig.NewMissingFileError(clusterConfig.Name, kubeconfigPath)
 	}
+
 	return nil
 }
 
 func (cc *createClusterOptions) createCluster(cmd *cobra.Command) error {
 	ctx := cmd.Context()
 
+	fmt.Fprintf(os.Stderr, "Create cluster run\n")
```
Comment (chrisdoherty4) [inline cmd/eksctl-anywhere/cmd/createcluster.go:91]: Remove

File: cmd/eksctl-anywhere/cmd/createcluster.go [hunk 2]
```diff
@@ -144,13 +152,14 @@ func (cc *createClusterOptions) createCluster(cmd *cobra.Command) error {
...
 			Name:           clusterSpec.Name,
-			KubeconfigFile: uc.kubeConfig(clusterSpec.Name),
+			KubeconfigFile: kubeconfig.FromClusterName(clusterSpec.Name),
 		},
 		ManagementCluster: cluster,
 		Provider:          deps.Provider,
 	}
 	createValidations := createvalidations.New(validationOpts)
 
+	fmt.Printf("Before run\n%+v\n%+v\n", cluster, validationOpts)
```
Comment (chrisdoherty4) [inline cmd/eksctl-anywhere/cmd/createcluster.go:162]: Remove

File: cmd/eksctl-anywhere/cmd/createcluster.go [hunk 3]
```diff
@@ -118,6 +125,7 @@ func (cc *createClusterOptions) createCluster(cmd *cobra.Command) error {
 		}
 	}
 
+	fmt.Printf("Before create\n")
```
Comment (chrisdoherty4) [inline cmd/eksctl-anywhere/cmd/createcluster.go:128]: Remove

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Pull request merged**